src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs

```diff
@@ -142,12 +160,25 @@ public async Task<T> GetConfigurationAsync(CancellationToken cancel)
                         _currentConfiguration = await _configRetriever.GetConfigurationAsync(MetadataAddress, _docRetriever, CancellationToken.None).ConfigureAwait(false);
                         _lastRefresh = now;
                         _syncAfter = DateTimeUtil.Add(now.UtcDateTime, AutomaticRefreshInterval);
+                        if (_configValidator != null)
+                        {
+                            ConfigurationValidationResult result = _configValidator.Validate(_currentConfiguration);
+                            if (!result.Succeeded)
+                            {
+                                throw new ConfigurationValidationException(LogMessages.IDX20810, result.Exception);
+                            }
+                        }
                     }
                     catch (Exception ex)
                     {
                         _syncAfter = DateTimeUtil.Add(now.UtcDateTime, AutomaticRefreshInterval < RefreshInterval ? AutomaticRefreshInterval : RefreshInterval);
                         if (_currentConfiguration == null) // Throw an exception if there's no configuration to return.
                             throw LogHelper.LogExceptionMessage(new InvalidOperationException(LogHelper.FormatInvariant(LogMessages.IDX20803, (MetadataAddress ?? "null")), ex));
+                        else if (ex is ConfigurationValidationException)
+                        {
+                            _currentConfiguration = null;
+                            throw LogHelper.LogExceptionMessage(ex);
```

Comment:
not sure we should throw here.
this clause is not needed if we use the logic above.